privval/grpc/client_test.go

```diff
@@ -0,0 +1,168 @@
+package grpc_test
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	grpc "google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	"github.com/tendermint/tendermint/libs/log"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+	tmgrpc "github.com/tendermint/tendermint/privval/grpc"
+	privvalproto "github.com/tendermint/tendermint/proto/tendermint/privval"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+const chainID = "chain-id"
+
+func dialer(pv types.PrivValidator, logger log.Logger) (*grpc.Server, func(context.Context, string) (net.Conn, error)) {
+	listener := bufconn.Listen(1024 * 1024)
+
+	server := grpc.NewServer()
+
+	s := tmgrpc.NewSignerServer(chainID, pv, logger)
+
+	privvalproto.RegisterPrivValidatorAPIServer(server, s)
+
+	go func() {
+		if err := server.Serve(listener); err != nil {
+			panic(err)
```

Comment:
does this mean the goroutine is going to panic during test cleanup when `srv.Stop()` is called?

Reply:
Nope, it will only panic when serve is called. I can exit the test if it fails as an alternative?